static/js/chart/draw.ts

```diff
@@ -737,6 +822,60 @@ function drawLineChart(
   return !hasFilledInValues;
 }
 
+/**
+ * Adds tooltip element within a given container.
+ *
+ * @param containerId
+ */
+function addTooltip(containerId: string): void {
```

Comment:
nit: these two utility functions should go to the top of the file (defined before they are referenced)

Reply:
done